### PR TITLE
Docker/MultiPart Stage Build

### PR DESCRIPTION
**Pull request by Awsaf780**
Fixes #

- [x] Does not depend on other PRs
- [ ] Depends on other PR being merged first: #

## What kind of change does this PR introduce (check at least one)

- [ ] Bugfix
- [ ] Improvement
- [x] New feature
- [ ] Refactor
- [ ] Infrastructure
- [ ] Other (please specify)

## How would you describe the changes in a changelog (1-2 lines)?
Serve react application with NGINX

## What was done
- Create docker-compose.prod file to configure NGINX 
- Change npm start to npm build
- Copy build files to nginx html directory
- Remove files other than build to decrease image size
- Expose on port 8080

## How to test

- [ ] Needs regression testing (add QA needed label)

## Deploy risk & follow up

Please assess risk & follow up on deployment of this PR. Elaborate your choice if selecting anything other than 'No risk'.

- [x] No risk
- [ ] Low
- [ ] High

## Reminder list

- Determine whether REST API Documentation should be updated
- Make an assessment on whether unit tests should be updated and/or added
- Ensure follow up tasks are linked to this PR

QA - if regression testing was requested:

- [ ] QA has verified

### CODE REVIEW

**Review by musfique-m — CHANGES_REQUESTED** [review]

File: react/todo-app/Dockerfile [hunk 1]
```diff
@@ -1,20 +1,20 @@
...
-
-# Command to run script
+# Use Base Image (Alpine used because it is lightweight)
+FROM node:alpine
+
+# Create and Specify working directory to organize our application into
+WORKDIR /app
+
+# Copy package.json first, because it defines project dependencies. Since it rarely changes, builds will be faster because of cache.
+COPY package.json .
```
[the remark this answers is not in the document]
Comment (Awsaf780) [reply react/todo-app/Dockerfile]: Resolved in commit [8133ca9](https://github.com/CraftsmenLtd/coe-batch-q322/pull/297/commits/8133ca9f0bad832b840d541eaa94838f73fbbe99)

File: react/todo-app/docker-compose.prod.yml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+version: '3.8'
+services:
+    react-nginx-app:
+      stdin_open: true
```
Comment (musfique-m) [inline react/todo-app/docker-compose.prod.yml:4]: Please remove this. We don't want to run our production container in interactive mode by default.

**Review by musfique-m — CHANGES_REQUESTED** [review]

File: react/todo-app/docker-compose.prod.yml (hunk 2 above)
Comment (Awsaf780) [reply react/todo-app/docker-compose.prod.yml]: Removed in commit [59fb32a](https://github.com/CraftsmenLtd/coe-batch-q322/pull/297/commits/59fb32ae0a5d6673249da0e92ade7f7c5267af00)

**Review by musfique-m — APPROVED** [review]
Looks good.

**Pull request merged**